bin/src/data/handlertorch.py

```diff
@@ -7,68 +7,63 @@
 from torch.utils.data import Dataset, DataLoader
 from torch.nn.utils.rnn import pad_sequence
 from .csv import CsvLoader
-from typing import Any, Tuple
+from typing import Any, Tuple, Union, Literal
 
 class TorchDataset(Dataset):
     """
     Class for creating a torch dataset
     """
-    def __init__(self, csvpath : str, experiment : Any) -> None:
-        self.csvpath = csvpath
-        self.parser = CsvLoader(experiment, csvpath)
+    def __init__(self, csvpath: str, experiment: Any, split: Tuple[None, int] = None) -> None:
+        self.input, self.label, self.meta, self.length = CsvLoader(experiment, csvpath, split=split).get_all_items_and_length()
+        self.input, self.label = self.convert_dict_to_dict_of_tensors(self.input), self.convert_dict_to_dict_of_tensors(self.label)
 
-    def convert_list_of_numpy_arrays_to_tensor(self, data: list) -> Tuple[torch.Tensor, torch.Tensor]:
+    def convert_to_tensor(self, data: Union[np.ndarray, list], transform_method: Literal['pad_sequences'] = 'pad_sequences', **transform_kwargs) -> Union[torch.tensor, list]:
```

Comment:
transform method is referring to padding. In the future are going to be there other methods? if so they will be handled all by the Literal[keywords] definition?

Reply:
Yes